File: Crawler/src/entities/Domain.ts
import { Entity, PrimaryColumn, ManyToOne, Column } from "typeorm";
import { DomainGroup } from "./DomainGroup";

@Entity("domains")
export class Domain {
  @PrimaryColumn({ update: false })
  origin: string;

  @Column({ update: false })
  domainGroupId: number;

  @Column()
  lastFastCrawlingUtc: Date;
 
  @Column()
  lastDeepCrawlingUtc: Date;

  @ManyToOne(() => DomainGroup, domainGroup => domainGroup.domains)
  domainGroup: DomainGroup;

  constructor(props?: Partial<Domain>) {
    Object.assign(this, props);
  }
}

File: QA.Search.Data/Models/Domain.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace QA.Search.Data.Models
{
    public class Domain
    {
        [Key]
        [MaxLength(450)]
        public string Origin { get; set; }

        public int DomainGroupId { get; set; }
        [Required]
        public DomainGroup DomainGroup { get; set; }

        
        public DateTime? LastFastCrawlingUtc { get; set; }
        public DateTime? LastDeepCrawlingUtc { get; set; }
    }
}
